### PR TITLE
Repo changelog support in auditquery

### PR DESCRIPTION
**Pull request by ligangty**
A very draft stub of UI for change log part, not support search yet. It will show all change log now in the cache. Server-side pagination with searching should be added in another patch.

### CODE REVIEW

File: cache/src/main/resources/infinispan.xml [hunk 1]
```diff
@@ -25,6 +25,28 @@
                 <property name="default.directory_provider">ram</property>
             </indexing>
         </local-cache>
+
+        <local-cache name="file_event_cache" configuration="local-template">
```
Comment (sswguo) [inline cache/src/main/resources/infinispan.xml]: This should be unused any more and i had moved the file event cache into infinispan-cluster.xml.

File: model-java/src/main/java/org/commonjava/auditquery/changelog/RepositoryChangeLog.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+import org.hibernate.search.annotations.Indexed;
+
+import javax.persistence.Entity;
+import java.io.Serializable;
+import java.util.Date;
+
+@Entity
+@Indexed
+public class RepositoryChangeLog
+	implements Serializable
```
Comment (sswguo) [inline model-java/src/main/java/org/commonjava/auditquery/changelog/RepositoryChangeLog.java]: We may need to use Externalizable as John suggested before: https://github.com/Commonjava/auditquery/pull/6#discussion_r274143791

File: cache/src/main/resources/infinispan.xml (hunk 1 above)
Comment (ligangty) [reply cache/src/main/resources/infinispan.xml]: ok.

File: model-java/src/main/java/org/commonjava/auditquery/changelog/RepositoryChangeLog.java (hunk 2 above)
Comment (ligangty) [reply model-java/src/main/java/org/commonjava/auditquery/changelog/RepositoryChangeLog.java]: will do.

File: model-java/src/main/java/org/commonjava/auditquery/changelog/RepositoryChangeLog.java [hunk 3]
```diff
@@ -22,226 +22,257 @@
...
+        }
+
+        return true;
+    }
+
+    @Override
+    public void readExternal( ObjectInput in )
+            throws IOException, ClassNotFoundException
+    {
+        this.storeKey = (String) in.readObject();
```
Comment (sswguo) [inline model-java/src/main/java/org/commonjava/auditquery/changelog/RepositoryChangeLog.java]: We may need to check the version here and see if that's compatible. But looks fine atm.

**Pull request merged**